### PR TITLE
📊  feat: Google tag manager integration

### PR DESCRIPTION
**Pull request by DenisPalnitsky**
## Summary
Adds optional Google Tag Manager integration.
Discussion:  https://github.com/danny-avila/LibreChat/discussions/2430

## Change Type

Please delete any irrelevant options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Translation update
- [ ] Documentation update

## Testing

Testing instructions (based on [this](https://analystadmin.medium.com/implementing-google-analytics-and-google-tag-manager-into-a-react-js-app-e986579cd0ee) article): 
1. Copy .env.example to .env file 
2. Uncomment `ANALYTICS_GTM_ID` in the .env file and set to to a valid Google Tag Manager account ID that has format similar to `GTM-AAAAA123`. You can enter random string there. In that case you won't see events in GTM but you still can test that functionality works. 
3. Run backend and frontend
4. Load application and click around 
5. In network tab you will see requests to gtm.js
image
6. Navigate to dev console and enter 'dataLayer' there. You should get list of events back 
image
7. (optionally) Configure GTM account as described in [GTM Set Up section here](https://analystadmin.medium.com/implementing-google-analytics-and-google-tag-manager-into-a-react-js-app-e986579cd0ee) and verify that events are received by GTM. 

## Checklist

Please delete any irrelevant options.

- [x] My code adheres to this project's style guidelines
- [x] I have performed a self-review of my own code
- [x] My changes do not introduce new warnings
- [x] Local unit tests pass with my changes
- [x] Any changes dependent on mine have been merged and published in downstream modules.

### CODE REVIEW

**Comment by DenisPalnitsky** [comment]
@danny-avila Could you please take a look at this short PR?

File: api/package.json [hunk 1]
```diff
@@ -86,6 +86,7 @@
     "passport-jwt": "^4.0.1",
     "passport-local": "^1.0.0",
     "pino": "^8.12.1",
+    "react-gtm-module": "^2.0.11",
```
Comment (danny-avila) [inline api/package.json]: this is not installed correctly. Please revert the installation commit or 

```bash
# uninstall from api workspace
cd api && npm uninstall react-gtm-module

# install in client workspace
cd ../client && npm install react-gtm-module

**Review by danny-avila — CHANGES_REQUESTED** [review]

**Comment by danny-avila** [comment]
Please also resolve the conflicts. it's best to undo the package installation altogether, then rebase latest main branch to your branch, then install correctly

**Comment by DenisPalnitsky** [comment]
> Please also resolve the conflicts. it's best to undo the package installation altogether, then rebase latest main branch to your branch, then install correctly

Done

**Comment by danny-avila** [comment]
@DenisPalnitsky do you mind adding documentation for this before I merge?

https://github.com/LibreChat-AI/librechat.ai

Also let me know if the changes I made work for you, note I rebased the branch.

**Comment by DenisPalnitsky** [comment]
@danny-avila here is the PR for documentation update https://github.com/LibreChat-AI/librechat.ai/pull/49
It's a brief description of ENV var. Let me know if you think that we need more info on that topic

**Pull request merged**